File: CucumberJVM/Chapter13/13.1-TheLocalScenarioContextPattern/After/src/main/java/com/wimp/localscenariocontext/app/services/NotificationService.java
package com.wimp.localscenariocontext.app.services;

import com.wimp.localscenariocontext.app.infrastructure.DataContext;
import com.wimp.localscenariocontext.app.models.Notification;

public class NotificationService {
    public static void sendCancellationNotification(String customerName) {
        DataContext.INSTANCE.saveNotification(new Notification(customerName, "Your order has been cancelled."));
    }

    public static boolean wasNotificationSent(String customerName) {
        return !DataContext.INSTANCE.getNotificationsByCustomerName(customerName).isEmpty();
    }
}

File: Reqnroll/Chapter13/13.2-TheShareableScenarioContextPattern/After/WIMP.ShareableScenarioContextSample.App/Services/NotificationService.cs
using WIMP.ShareableScenarioContextSample.App.Infrastructure;
using WIMP.ShareableScenarioContextSample.App.Models;

namespace WIMP.ShareableScenarioContextSample.App.Services;

public static class NotificationService
{
    public static void SendCancellationNotification(string customerName)
    {
        DataContext.Instance.SaveNotification(new Notification(customerName, "Your order has been cancelled."));
    }

    public static bool WasNotificationSent(string? customerName)
    {
        return DataContext.Instance.GetNotificationsByCustomerName(customerName ?? "n/a").Any();
    }
}
